source/slang/slang-compiler.cpp

```diff
@@ -2483,13 +2483,14 @@ SlangResult dissassembleDXILUsingDXC(
         BackEndCompileRequest* compileRequest,
         EndToEndCompileRequest* endToEndReq)
     {
-        // If we are about to generate output code, but we still
+        // When dynamic dispatch is disabled, the program must
+        // be fully specialized by now. So we check if we still
         // have unspecialized generic/existential parameters,
-        // then there is a problem.
+        // and report them as an error.
         //
         auto program = compileRequest->getProgram();
         auto specializationParamCount = program->getSpecializationParamCount();
-        if( specializationParamCount != 0 )
+        if (compileRequest->disableDynamicDispatch && specializationParamCount != 0)
```

Comment:
This works for the two extreme cases:

* All type parameters have been specialized
* *No* type parameters have been specialized

The missing piece at that point is some kind of partial specialization. I don't think that the code at this site needs to change at all to support partial specialization, but we do need to figure out what it means for the public API that supports specialization.

My take is that akin to how we can specialize a parameter to a single type right now, we eventually need a way to "specialize" a parameter to a value representing:

* "Do not specialize this parameter; use dynamic dispatch instead"
* "Treat this parameter as if it could be one of these N types, but nothing else"